### ISSUE TITLE
Processing.sound only working first time a sketch is run

### ISSUE THREAD

**Issue by Richard-Mullan** [jdf/Processing.py-Bugs#260]
Hi, I have a weird problem, I'm not sure if it's a bug or if I'm doing something wrong, but I've had a look around and can't find anyone else mentioning it.
When I first open a sketch, it works perfectly as intended, I hear a nice beating tone. However if I exit and then run the program again, no sound is ever heard. It only works again if I close and reopen processing, regardless of any changes made or not made to the code, ie: making it interactive or anything works perfectly fine so long as it's the first time the sketch has been run.

TL;DR: sound is only audible the first time running a sketch every time processing is opened
Any help greatly appreciated, thanks.

```
add_library('sound')
def setup():
    global ch_1, ch_2
    size(800,480)
    ch_1 = SinOsc(this)
    ch_1.play(400,1)
    ch_2 = SinOsc(this)
    ch_2.play(401,1)
    
def draw():
    global ch_1, ch_2
    background(255)
```

**Comment by Richard-Mullan** [comment]
Another example of code which works perfectly well. Only reason this is noteworthy is that if the mouse is clicked, then released, then clicked again, the program works as intended, with the sound stopping and starting as it should. It's only when the sketch is closed and then run again without closing and re-opening the whole processing environment that there is an issue.

```
add_library('sound')

def setup():
    global ch_1, ch_2
    size(800,480)
    ch_1 = SinOsc(this)
    ch_1.play(0,1)
    ch_2 = SinOsc(this)
    ch_2.play(0,1)
    
def draw():
    global ch_1, ch_2
    background(255)
    
    if mousePressed:
        ch_1.set(400, 1, 0, 0)
        ch_2.set(401, 1, 0, 0)
    
    else:
        ch_1.set(0, 1, 0, 0)
        ch_2.set(0, 1, 0, 0)
```

**Comment by jdf** [comment]
Thanks for this bug. It looks to me like something in the sound library is incompatible with an environment that doesn't simply exit when a sketch finishes, which means it will have to be addressed in the Processing project. I'll look at that when I can, but it's probably not something I can address in Python Mode per se.

**Comment by jdf** [comment]
The sound library's `Engine` class creates a global singleton, which never goes away.

**Comment by Richard-Mullan** [comment]
Ah, interesting. Would there then be a way of re-instantiating the Engine
in setup every time the code is run? I've had a go but can't make it work.
Do you know what the Engine object is named by default? It must exist in
the background so I assume it has a name, but I can't see that information
on the github page.

Thanks for your help.

**Comment by jdf** [comment]
My plan is to submit a patch that adds a `teardown()` or `reset()` function to Engine. While waiting for that to be released, I'll try to use reflection to null out that singleton from Python Mode. It may take me some time to get to it; I'm sorry for the inconvenience!

**Comment by Richard-Mullan** [comment]
Don't apologise! I'm amazed such a minor bug has even made it to your
attention so quickly.

Let me just say this is hands down the best technical assistance I've ever
received. Yet another reason to love Processing.

Hope you have a good day.
Richard

**Comment by jdf** [comment]
Pushed in build 3052.
